File: Data-Structure-I/653-Two-Sum-IV-Input-is-a-BST.php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {
    private $s = [];
    /**
     * @param TreeNode $root
     * @param Integer $k
     * @return Boolean
     */
    function findTarget($root, $k) {
        if($root==NULL) return false;
        if(in_array($k - $root->val, $this->s)) return true;
        else $this->s[] = $root->val;
        return $this->findTarget($root->left, $k) || $this->findTarget($root->right, $k);
    }
}

File: Data-Structure-I/653-Two-Sum-IV-Input-is-a-BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        self.s = set()
        return self.check(root, k)
    def check(self, node, k):
        if not node:
            return False
        x = k - node.val
        if x in self.s:
            return True
        else:
            self.s.add(node.val)
        return self.check(node.left, k) or self.check(node.right, k)
        
